pallets/gear-eth-bridge/src/tests.rs

```diff
@@ -0,0 +1,642 @@
+    })
+}
+
+#[test]
+fn bridge_is_paused_err() {
+    init_logger();
+    new_test_ext().execute_with(|| {
+        const ERR: Error = Error::BridgeIsPaused;
+
+        run_to_block(42);
+        run_block_and_assert_messaging_error(
+            Request::SendEthMessage {
+                destination: H160::zero(),
+                payload: vec![],
+            },
+            ERR,
+        );
+    })
+}
+
+#[test]
+fn bridge_max_payload_size_exceeded_err() {
+    init_logger();
+    new_test_ext().execute_with(|| {
+        const ERR: Error = Error::MaxPayloadSizeExceeded;
+
+        run_to_block(42);
+
+        assert_ok!(GearEthBridge::unpause(RuntimeOrigin::root()));
+
+        let max_payload_size: u32 = <Test as crate::Config>::MaxPayloadSize::get();
+
+        run_block_and_assert_messaging_error(
+            Request::SendEthMessage {
+                destination: H160::zero(),
+                payload: vec![0; max_payload_size as usize + 1],
+            },
+            ERR,
+        );
+    })
+}
+
+#[test]
+fn bridge_queue_capacity_exceeded_err() {
+    init_logger();
+    new_test_ext().execute_with(|| {
+        const ERR: Error = Error::QueueCapacityExceeded;
+
+        run_to_block(42);
+
+        assert_ok!(GearEthBridge::unpause(RuntimeOrigin::root()));
+
+        for _ in 0..<Test as crate::Config>::QueueCapacity::get() {
+            assert_ok!(GearEthBridge::send_eth_message(
+                RuntimeOrigin::signed(SIGNER),
+                H160::zero(),
+                vec![]
+            ));
+        }
+
+        run_block_and_assert_messaging_error(
+            Request::SendEthMessage {
+                destination: H160::zero(),
+                payload: vec![],
+            },
+            ERR,
+        );
+    })
+}
+
+#[test]
+fn bridge_incorrect_value_applied_err() {
+    init_logger();
+    new_test_ext().execute_with(|| {
+        const ERR: Error = Error::IncorrectValueApplied;
+
+        run_to_block(42);
+
+        assert_ok!(GearEthBridge::unpause(RuntimeOrigin::root()));
+
+        let response = run_block_with_builtin_call(
+            SIGNER,
+            Request::SendEthMessage {
+                destination: H160::zero(),
+                payload: vec![],
+            },
+            1,
+        );
+
+        assert_eq!(
+            String::from_utf8_lossy(&response),
+            format!("Panic occurred: {}", builtin::error_to_str(&ERR))
+        );
+    })
+}
+
+mod utils {
+    use crate::builtin;
+
+    use super::*;
+    use gear_core::message::UserMessage;
+    use gprimitives::{ActorId, MessageId};
+
+    pub(crate) fn builtin_id() -> ActorId {
+        GearBuiltin::generate_actor_id(3)
+    }
+
+    #[track_caller]
+    pub(crate) fn run_block_and_assert_bridge_error(error: Error) {
+        assert_noop!(GearEthBridge::pause(RuntimeOrigin::root()), error.clone());
+
+        assert_noop!(GearEthBridge::unpause(RuntimeOrigin::root()), error.clone());
```

Comment:
Why do you need to pause/unpause it?

Reply:
to check storage noop, just normal practice